src/components/Patient/PatientRegistration.tsx

```diff
@@ -0,0 +1,1076 @@
+
+  const mutationFields: (keyof PatientModel)[] = [
+    "name",
+    "phone_number",
+    "emergency_phone_number",
+    "gender",
+    "blood_group",
+    "date_of_birth",
+    "age",
+    "address",
+    "permanent_address",
+    "pincode",
+    "nationality",
+    "state",
+    "district",
+    "local_body",
+    "ward",
+    "village",
+    "meta_info",
+    "ration_card_category",
+  ];
+
+  const mutationData: Partial<PatientModel> = {
+    ...Object.fromEntries(
+      Object.entries(form).filter(([key]) =>
+        mutationFields.includes(key as keyof PatientModel),
+      ),
+    ),
+    date_of_birth:
+      ageDob === "dob" ? dateQueryString(form.date_of_birth) : undefined,
+    year_of_birth: ageDob === "age" ? form.year_of_birth : undefined,
+    is_active: true,
+    is_antenatal: false,
+    passport_no: form.nationality === "Indian" ? form.passport_no : undefined,
+    meta_info: {
+      ...(form.meta_info as any),
+      occupation:
+        form.meta_info?.occupation === ""
+          ? undefined
+          : form.meta_info?.occupation,
+    },
+  };
+
+  const createPatientMutation = useMutation(routes.addPatient, {
+    body: { ...mutationData, facility: facilityId, ward_old: undefined },
+    onResponse: (resp) => {
+      if (resp.error) {
+        Notification.Error({
+          msg: t("patient_registration_error"),
+        });
+      } else {
+        Notification.Success({
+          msg: t("patient_registration_success"),
+        });
+        navigate(
+          `/facility/${facilityId}/patient/${resp.data?.id}/consultation`,
+        );
+      }
+    },
+  });
+
+  const updatePatientMutation = useMutation(routes.updatePatient, {
+    pathParams: { id: patientId || "" },
+    body: { ...mutationData, ward_old: undefined },
+    onResponse: (data) => {
+      if (data.error) {
+        Notification.Error({
+          msg: t("patient_update_error"),
+        });
+      } else {
+        Notification.Success({
+          msg: t("patient_update_success"),
+        });
+        goBack();
+      }
+    },
+  });
+
+  const patientQuery = useQuery({
+    queryKey: ["patient", patientId],
+    queryFn: query(routes.getPatient, {
+      pathParams: { id: patientId || "" },
+    }),
+    enabled: !!patientId,
+  });
+
+  const setAddress = async (args: {
+    state: (typeof form)["state"];
+    district?: (typeof form)["district"];
+    local_body?: (typeof form)["local_body"];
+    ward?: string;
+  }) => {
+    const { state, district, local_body, ward } = args;
+    setForm((f) => ({
+      ...f,
+      state,
+    }));
+    await new Promise((resolve) => setTimeout(resolve, 500));
+    const districts = await districtsQuery.refetch();
+
+    const matchedDistrict = districts.data?.find((d) => d.id === district);
+    if (!matchedDistrict) return;
+    setForm((f) => ({
+      ...f,
+      district: matchedDistrict.id,
+    }));
+
+    if (local_body) {
+      await new Promise((resolve) => setTimeout(resolve, 500));
```

Comment:
Why do we have a 500ms Wait here?

Reply:
Hack to wait for the state to complete. Another option would have been to redefine the queries as mutations, that would be complex and redundant 😅